src/units/abilities.cpp

```diff
@@ -1168,6 +1168,47 @@ void attack_type::modified_attacks(unsigned & min_attacks,
 	}
 }
 
+//Functions used for change damage_type list with damage
+static std::vector<std::string> damage_type_list(const unit_ability_list& abil_list, const std::string& type)
+{
+	std::vector<std::string> type_list;
+	for(auto& i : abil_list) {
+		if(!(*i.ability_cfg)[type].str().empty()){
+			type_list.push_back((*i.ability_cfg)[type].str());
+		}
+	}
+	if(type_list.size() == 2){
+		std::sort(type_list.begin(), type_list.end());
+	} else if(type_list.size() >= 3){
+		std::unordered_map<std::string, unsigned int> type_count;
+		for( const std::string& character : type_list ){
+			type_count[character]++;
+		}
+		std::sort( std::begin( type_list ) , std::end( type_list ) , [&]( const std::string& rhs , const std::string& lhs ){
+			return type_count[lhs] < type_count[rhs];
+		});
```

Comment:
Does this sort correctly if there are two damage types both with two instances? For example:
```
[damage]
type=cold
[/damage]
[damage]
type=cold
[/damage]
[damage]
type=arcane
[/damage]
[damage]
type=arcane
[/damage]
```
?

Reply:
that's why I implemented the loops if one inside the other, moreover the alphabetical classification must be done when the vector contains two or more elements, not simply two elements. I need to revert the changes you requested on this to resolve this.

Reply:
change reverted.
by first carrying out the alphabetical reorganization we ensure that if two cold and two fire exist in the vector cold are already placed before fire before the numerical sorting, and if these two elements have the greatest number of occurrences in the vector then logically all cold should be placed at the head of the vector before fire is classified in turn. The relative alphabetical order of the elements with the greatest number of copies should then be preserved.

Reply:
Alright, my bad.